pkg/controller/operators/olm/operatorgroup.go

```diff
@@ -411,9 +435,12 @@ func (a *Operator) updateNamespaceList(op *v1alpha2.OperatorGroup) ([]string, er
 			}
 			namespaceSet[ns] = struct{}{}
 		}
-	} else if selector == nil || selector.Empty() {
-		namespaceSet[corev1.NamespaceAll] = struct{}{}
 	} else {
+		if selector == nil || selector.Empty() {
+			selector = labels.Everything()
+			// set is sorted after returning, so being at the beginning can be relied upon
```

Comment:
not sure I follow this comment - where are we using order of the set? (asking because maps don't retain any ordering)

Reply:
Right, I should have specifically said slice instead. It's after being converted from a set to a slice in updateNamespaceList.